source/AMDSClientUi.cpp

```diff
@@ -0,0 +1,332 @@
+#include "AMDSClientUi.h"
+
+#include <QtGui>
+#include <QtNetwork>
+#include <QDateTimeEdit>
+#include <QDialog>
+
+#include "source/appController/AMDSClientAppController.h"
+#include "source/Connection/AMDSServer.h"
+#include "source/ClientRequest/AMDSClientRequest.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+#include "source/util/AMDSErrorMonitor.h"
+
+AMDSClientUi::AMDSClientUi(QWidget *parent) :
+    QDialog(parent)
+{
+	/// ==== server section ====
+	hostLabel_ = new QLabel(tr("&Server name:"));
+	portLabel_ = new QLabel(tr("&erver port:"));
+
+	hostLineEdit_ = new QLineEdit("10.51.3.5");
+	portLineEdit_ = new QLineEdit("28044");
+	portLineEdit_->setValidator(new QIntValidator(1, 65535, this));
+
+	hostLabel_->setBuddy(hostLineEdit_);
+	portLabel_->setBuddy(portLineEdit_);
+
+	activeServerComboBox_ = new QComboBox;
+	activeServerComboBox_->setFixedWidth(200);
+	activeServerComboBox_->addItem("");
+	connect(activeServerComboBox_, SIGNAL(currentIndexChanged(QString)), this, SLOT(onActiveServerChanged(QString)));
+
+	statusLabel_ = new QLabel(tr("This examples requires that you run the Acquaman Data Server example as well."));
+
+	connect(hostLineEdit_, SIGNAL(textChanged(QString)), this, SLOT(enableRequestDataButton()));
+	connect(portLineEdit_, SIGNAL(textChanged(QString)), this, SLOT(enableRequestDataButton()));
```

Comment:
There seems to be a little issue with these regarding the logic for setting the enabled state of the request data button.
When setting the IP or port to connect to initially there is no current text in the active server combo box (you've not connected yet), and thus the button is disabled. 
If you don't require a network session (as per line 69 of the app controller) none of the session signals will be fired and the button will only be re-enabled if you manually jump into either line edits after connecting and change a value.